titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/validator/JobIamValidator.java

```diff
@@ -87,7 +87,10 @@ public JobIamValidator(JobSecurityValidatorConfiguration configuration, IamConne
                 // populate the set with a specific error.
                 .thenReturn(Collections.<ValidationError>emptySet())
                 .onErrorResume(throwable -> {
-                    registry.counter(validationFailureId.withTag(failedMetricIamTag, iamRoleName)).increment();
+                    registry.counter(validationFailureId
+                            .withTag(failedMetricIamTag, iamRoleName)
+                            .withTag(metricReasonTag, throwable.getMessage()))
```

Comment:
Is it safe to store exception message as a metric tag value?

Reply:
I prefer to put the Exception class name here generally if the message contains data to reduce the cardinality.

Reply:
The Exception class name here will always be an `IamConnectorException`. That class controls the type/size of the message, which is why I used it. If we were really have concerns about it, that class has a `ErrorType` that we could use, though it's less detailed.

Reply:
`ErrorType` sounds much better. Details can be written in the log file.

Reply:
Will do.